plugins/hypervisors/xenserver/src/com/cloud/hypervisor/xenserver/resource/Xenserver625StorageProcessor.java

```diff
@@ -538,7 +562,7 @@ public Answer backupSnapshot(final CopyCommand cmd) {
                     final String[] tmp = result.split("#");
                     snapshotBackupUuid = tmp[0];
                     physicalSize = Long.parseLong(tmp[1]);
-                    finalPath = folder + File.separator + snapshotBackupUuid;
+                    finalPath = folder + File.separator + snapshotBackupUuid + ".vhd";
```

Comment:
This change will result in a ".vhd" extension being written to snapshot_store_ref as this is being returned in the CopyCmdAnswer. Although I prefer this, the implication of this change is that there will be inconsistency in snapshot names in a setup upgraded to this version. I am wondering and please confirm that this is not going to cause the UI to not show the older snapshots, only thing will be that the older snapshots' install_path will be without extension. For consistency can be modify the upgrade script to append ".vhd" to all the old Xen based snapshots ?

cc @rhtyd @jburwell